JavaScript/Embeddings/SQL (JS template).sublime-syntax

```diff
@@ -0,0 +1,18 @@
+%YAML 1.2
+---
+# http://www.sublimetext.com/docs/syntax.html
+# highlight tagged template strings
+scope: source.sql.js-template
+version: 2
+hidden: true
+
+extends: Packages/SQL/SQL.sublime-syntax
+
+contexts:
+  prototype:
+    - meta_prepend: true
+    - include: scope:source.js#text-interpolations
+
+  strings-content:
+    - meta_prepend: true
+    - include: scope:source.js#string-interpolations
```

Comment:
Current SQL syntax doesn't use this context. It wouldn't have any effect.

It doesn't support `meta.string string` vs. `meta.string meta.interpolation` in general.

Reply:
`string-interpolations` removes a single scope from the context it is included to.

This is primarily required as we want to clear `string` scope in interpolations.

```
"Text { interpolated.variable } !"
^^^^^^ meta.string string - meta.interpolation
      ^^^^^^^^^^^^^^^^^^^^^^^^^ meta.string meta.interpolation - string
                               ^^^ meta.string string - meta.interpolation
```

That's required, because `string` is treated special but we want interpolations to be treated as normal source code. And we don't want it to be highlighted like a string. Any unmatched token in interpolations would be highlighted with string color, which may look odd and broken.

Therefore "modern" syntax definitions exclude `prototype` context from strings to enforce special `string-interpolations` contexts to be added by inheriting syntaxes.

Interpolation in all other non-string contexts is handled by `text-interpolations` which does not clear any scopes.